src/main/java/org/folio/template/dao/TemplateDaoImpl.java

```diff
@@ -30,61 +32,61 @@ public TemplateDaoImpl(Vertx vertx, String tenantId) {
 
   @Override
   public Future<List<Template>> getTemplates(String query, int offset, int limit) {
-    Future<Results<Template>> future = Future.future();
+    Promise<Results<Template>> promise = Promise.promise();
     try {
       String[] fieldList = {"*"};
       CQLWrapper cql = getCQL(query, limit, offset);
-      pgClient.get(TEMPLATES_TABLE, Template.class, fieldList, cql, true, false, future.completer());
+      pgClient.get(TEMPLATES_TABLE, Template.class, fieldList, cql, true, false, promise);
     } catch (Exception e) {
-      future.fail(e);
+      promise.fail(e);
     }
-    return future.map(Results::getResults);
+    return promise.future().map(Results::getResults);
   }
 
   @Override
   public Future<Optional<Template>> getTemplateById(String id) {
-    Future<Results<Template>> future = Future.future();
+    Promise<Results<Template>> promise = Promise.promise();
     try {
-      Criteria idCrit = new Criteria(TEMPLATE_SCHEMA_PATH);
+      Criteria idCrit = new Criteria();
       idCrit.addField(TEMPLATES_ID_FIELD);
       idCrit.setOperation("=");
-      idCrit.setValue(id);
-      pgClient.get(TEMPLATES_TABLE, Template.class, new Criterion(idCrit), true, future.completer());
+      idCrit.setVal(id);
+      pgClient.get(TEMPLATES_TABLE, Template.class, new Criterion(idCrit), true, promise);
     } catch (Exception e) {
-      future.fail(e);
+      promise.fail(e);
     }
-    return future
+    return promise.future()
       .map(Results::getResults)
       .map(templates -> templates.isEmpty() ? Optional.empty() : Optional.of(templates.get(0)));
   }
 
   @Override
   public Future<String> addTemplate(Template template) {
-    Future<String> future = Future.future();
-    pgClient.save(TEMPLATES_TABLE, template.getId(), template, future.completer());
-    return future;
+    Promise<String> promise = Promise.promise();
+    pgClient.save(TEMPLATES_TABLE, template.getId(), template, promise);
+    return promise.future();
   }
 
   @Override
   public Future<Boolean> updateTemplate(Template template) {
-    Future<UpdateResult> future = Future.future();
+    Promise<UpdateResult> promise = Promise.promise();
     try {
       Criteria idCrit = new Criteria();
       idCrit.addField(TEMPLATES_ID_FIELD);
       idCrit.setOperation("=");
-      idCrit.setValue(template.getId());
-      pgClient.update(TEMPLATES_TABLE, template, new Criterion(idCrit), true, future.completer());
+      idCrit.setVal(template.getId());
+      pgClient.update(TEMPLATES_TABLE, template, new Criterion(idCrit), true, promise);
```

Comment:
Can you use `update` that directly takes the `id`?

    pgClient.update(TEMPLATES_TABLE, template, template.getId(), promise);